tests/smoke-tests/test-block-weights.ts

```diff
@@ -0,0 +1,290 @@
+}
+
+interface BlockLimits {
+  normal: BN;
+  operational: BN;
+}
+
+describeSmokeSuite(
+  `Verify weights of blocks in the past ${(timePeriod / (1000 * 60 * 60)).toFixed(2)} hours`,
+  { wssUrl, relayWssUrl },
+  (context) => {
+    before("Retrieve all weight limits and usage", async function () {
+      this.timeout(240000);
+
+      const signedBlock = await context.polkadotApi.rpc.chain.getBlock(
+        await context.polkadotApi.rpc.chain.getFinalizedHead()
+      );
+
+      const lastBlockNumber = signedBlock.block.header.number.toNumber();
+      const lastBlockTime = getBlockTime(signedBlock);
+
+      const firstBlockTime = lastBlockTime - timePeriod;
+      debug(`Searching for the block at: ${new Date(firstBlockTime)}`);
+      const firstBlockNumber = (await limiter.wrap(fetchHistoricBlockNum)(
+        context.polkadotApi,
+        lastBlockNumber,
+        firstBlockTime
+      )) as number;
+
+      const length = lastBlockNumber - firstBlockNumber;
+      const blockNumArray = Object.freeze(Array.from({ length }, (_, i) => firstBlockNumber + i));
+      const limits = context.polkadotApi.consts.system.blockWeights;
+
+      const getLimits = async (blockNum: number) => {
+        const blockHash = await context.polkadotApi.rpc.chain.getBlockHash(blockNum);
+        const apiAt = await context.polkadotApi.at(blockHash);
+        const specVersion = apiAt.consts.system.version.specVersion.toNumber();
+        const events = await apiAt.query.system.events();
+
+        if (specVersion >= 1700) {
+          const { normal, operational, mandatory } = await apiAt.query.system.blockWeight();
+          return {
+            blockNum,
+            hash: blockHash.toString(),
+            weights: {
+              normal,
+              operational,
+              mandatory,
+            },
+            events,
+          };
+        }
+      };
+
+      const results = await Promise.all(
+        blockNumArray.map((num) => limiter.schedule(() => getLimits(num)))
+      );
+
+      context.storeMemo("blockLimits", {
+        normal: new BN(limits.perClass.normal.maxTotal.toJSON() as number),
+        operational: new BN(limits.perClass.operational.maxTotal.toJSON() as number),
+      });
+      context.storeMemo("blockInfoArray", results);
+    });
+
+    // This test is more for verifying that the test code is correctly returning good quality data
+    // that the rest of the test suite performs verification on
+    it("should be returning unique block hashes in array", async () => {
+      const hashes = context.getMemo("blockInfoArray").map((a) => a.hash);
+      const set = new Set(hashes);
+      expect(hashes.length, "Duplicate hashes in retrieved data, investigate test").to.be.equal(
+        set.size
+      );
+    });
+
+    // Normal class
+    it("normal usage should be less than normal dispatch class limits", async function () {
+      const blockLimits: BlockLimits = context.getMemo("blockLimits");
```

Comment:
I'm not a big fan of using a generic storage within the same test set, for few reasons:
- It hides the variables you store (when writing a new `it` test, you need to read all the others to see if there are some memos)
- It hides the type of variable (you can store whatever your want and even modify it)
- It can easily be replaced by a variable in your set that would be easily to find (all variable in the set are at the beginning) and properly typed.

Reply:
This is pretty standard way of dealing with test data though, because the alternative is that you either:
- Replicate the same setup to prepare your data, n times, where n is number of tests in your suite
- Rely on global variables 

Memoizing data is the standard way of moving data around with your context, you don't lose types.

Reply:
Killed the memo's dead, have consigned myself to live and let `let` live.